components/api/src/manager.rs

```diff
@@ -72,6 +74,51 @@ pub struct ApiManagerConfig {
     pub server_restart_retry_timeout: u64,
     /// The attempts counts of restarting HTTP-server in case the server failed to restart.
     pub server_restart_max_retries: u16,
+    /// Disables signal handling for HTTP servers. By default, the HTTP servers will shut down
+    /// on receiving SIGINT, SIGTERM or SIGQUIT, and will also set the handler and ignore
+    /// SIGHUP.
+    ///
+    /// This setting is a no-op on platforms not based on Unix.
+    pub disable_signals: bool,
+}
+
+impl ApiManagerConfig {
+    /// Creates a new config with the specified servers and API aggregator. Other parameters are
+    /// set to reasonable defaults.
+    pub fn new(
+        servers: HashMap<ApiAccess, WebServerConfig>,
+        api_aggregator: ApiAggregator,
+    ) -> Self {
+        let mut this = Self::default();
+        this.servers = servers;
+        this.api_aggregator = api_aggregator;
+        this
+    }
+
+    /// Sets the retry policy for this config.
+    pub fn with_retries(mut self, timeout: u64, max_retries: u16) -> Self {
+        self.server_restart_retry_timeout = timeout;
+        self.server_restart_max_retries = max_retries;
+        self
+    }
+
+    /// Disables signal handling.
+    pub fn disable_signals(mut self) -> Self {
```

Comment:
Does it make sense to add the prefix `with_` here ?

Reply:
I used the same naming [as used in `actix_web`](https://docs.rs/actix-web/2.0.0/actix_web/struct.HttpServer.html#method.disable_signals).